next/components/pages/DocumentsPage.tsx

```diff
@@ -1,214 +1,129 @@
-  const [documentData, setDocumentData] = React.useState<DocumentResponse>({
-    documents: [],
-    fileCategories: [],
-    count: 0,
-  })
-
-  const SORT_OPTIONS: SortOption[] = [
-    { key: 'desc', title: t('sort_desc') },
-    { key: 'asc', title: t('sort_asc') },
-  ]
-
-  const resultsRef = React.useRef<HTMLDivElement>(null)
-
-  const [fetchingData, setFetchingData] = React.useState(true)
-  const [offsetPage, setOffsetPage] = React.useState(1)
-  const [sort, setSort] = React.useState(SORT_OPTIONS[0])
-  const [query, setQuery] = React.useState('')
-  const [visibleQuery, setVisibleQuery] = React.useState('')
 
-  const noOfPages = Math.ceil(documentData.count / DOCUMENTS_LIMIT)
+  // TODO add scroll to results
 
-  const scrollToResults = React.useCallback(() => {
-    resultsRef.current?.scrollIntoView({
-      behavior: 'smooth',
-      block: 'nearest',
-      inline: 'start',
-    })
-  }, [resultsRef])
+  const [filters, setFilters] = useState<DocumentsFilters>(documentsDefaultFilters)
 
-  React.useEffect(() => {
-    const fetchDocuments = async () => {
-      const res = await fetch(
-        `/api/documents?offset=${(offsetPage - 1) * DOCUMENTS_LIMIT}&sort=date_added:${
-          sort.key
-        }&query=${query}`
-      )
-      const data: DocumentResponse = await res.json()
+  const { input, setInput, searchValue, setSearchValue } = useSearch({ syncWithUrlQuery: false })
 
-      setDocumentData(data)
-      setFetchingData(false)
-    }
-
-    fetchDocuments()
-  }, [offsetPage, sort, query])
+  const { dataToDisplay, loadingAndNoDataToDisplay } = useSwrWithExtras(
+    getDocumentsSwrKey(filters),
+    documentsFetcher(filters)
+  )
 
-  const handleChangeOffsetPage = (num: number) => {
-    if (num > 0 && num <= noOfPages) setOffsetPage(num)
+  const handlePageChange = (page: number) => {
+    setFilters({ ...filters, page })
   }
 
-  const handleSearch = React.useCallback(() => {
-    setQuery(visibleQuery)
-    scrollToResults()
-  }, [setQuery, visibleQuery, scrollToResults])
-
-  const handleSearchReset = React.useCallback(() => {
-    setQuery('')
-    setVisibleQuery('')
-  }, [setQuery, setVisibleQuery])
+  const handleCategoryChange = (categoryId: string | null) => {
+    setFilters({ ...filters, page: 1, categoryId })
+  }
 
-  const onSearchBarKeyPress = React.useCallback(
-    (event: React.KeyboardEvent<HTMLInputElement>) => {
-      if (event.key === 'Enter') {
-        handleSearch()
-      }
-    },
-    [handleSearch]
-  )
+  const handleSortChange = (sort: Sort) => {
+    setFilters({ ...filters, sort })
+  }
 
-  const onSearchBarRightIconKeyPress = React.useCallback(
-    (event: React.KeyboardEvent<HTMLInputElement>) => {
-      if (event.key === 'Enter') {
-        handleSearchReset()
-      }
-    },
-    [handleSearchReset]
-  )
+  useEffect(() => {
+    setFilters({ ...filters, searchValue })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchValue])
```

Comment:
might be nicer to use the calback version of setState and avoid the eslint-ignore:

```suggestion
useEffect(() => {
    setFilters((prevFilters) => ({ ...prevFilters, searchValue }))
  }, [searchValue])
```

Reply:
Yes, for sure :) thx